.pipelines/common.yml

```
@@ -77,3 +80,9 @@ jobs:
 
       - script: ./run_functional_tests.sh ${{ parameters.FunctionalTestArguments }}
         displayName: Run Functional Tests
+
+      - ${{ if parameters.PublishPerfResults }}:
+        - task: PublishPipelineArtifact@1
+          inputs:
+            artifactName: "perf-results"
```

Looking at the DevOps page, it seems all the jobs in the pipeline are writing to the same artifact name and overwriting each other. I think the `perf-results` name is going to need to be parametrized.

Maybe this works?
```suggestion
            artifactName: "${{ parameters.Name }}-perf-results"
```

---

Yeah, for now there's only a single job that runs the perf test, but you're right, we'll probably extend that to more jobs in the future.

---

Oh I hadn't even realized that. I wonder what happens if multiple jobs write the same artifact then? Does it throw an error?

---

Yes, you can't override.